test/common/apiHelpers.js

```diff
@@ -169,25 +169,20 @@ function getBlocksToWaitPromise () {
 		});
 }
 
-function waitForConfirmations (transactions, limitHeight = 10) {
+function waitForConfirmations (transactions, limitHeight) {
+	limitHeight = limitHeight || 10;
 
 	function checkConfirmations (transactions) {
 		return node.Promise.map(transactions, function (transaction) {
 			return getTransactionPromise(transaction);
 		})
 			.then(function (res) {
 				return node.Promise.each(res, function (result) {
-					node.expect(result).to.have.property('success').to.be.ok;
-					node.expect(result).to.have.property('transaction').not.empty;
-				})
-					.then(true)
-					.catch(function (err) {
-						throw err;
-					});	
-			})
-			.catch(function (err) {
-				throw err;
-			});	
+					if (result.success === false) {
+						throw Error(result.error);
```

Comment:
Why not something like: `node.expect(result).to.have.property('success').to.be.ok;`?

Reply:
@MaciejBaj suggested to use code instead of messing around with Mocha code. I am ok with both solutions.